controller/src/main/java/io/pravega/controller/store/stream/ZKStoreHelper.java

```diff
@@ -62,6 +64,34 @@ public ZKStoreHelper(final CuratorFramework cf, Executor executor) {
         return getChildren("/store");
     }
 
+    CompletableFuture<List<CuratorTransactionResult>> transaction(List<CuratorOp> opList) {
+        final CompletableFuture<List<CuratorTransactionResult>> result = new CompletableFuture<>();
+        try {
+            // We don't know the accurate path on which this transaction failed.
+            // See also ZOOKEEPER-3290
+            String msg = getTransactionErrorMessage(opList);
```

Comment:
If cycles matter I'd like to set parameter `path` to `<unknown>` or `<unknown-in-transaction>` because we have to pass a string to `callback` when call it and, in another perspective, this message is not a `path`.